### PR TITLE
hdf4: For compatibility with netcdf, remove bin/ncgen

### LINKED ISSUE #1325: Conflict with netcdf because ncgen is not deleted
I'm installing hdf4 but it conflicts with previously installed netcdf package. When I ran "brew doctor" afterwards it says hdf4 is not linked. When I try to link, it says it cannot do it because it conflicts with /usr/local/bin/ncgen which is provided by netcdf. 

I noticed in the hdf4.rb formula that this was already accounted for, as it includes:

  
 "-DHDF4_ENABLE_NETCDF=OFF", # Conflict. Just install NetCDF for this.

However, even with the option not to build netcdf, the hdf4 package still creates two executables: ncgen and ncdump. This is partly fixed by the hdf4.rb formula here:

mkdir "build" do
      system "cmake", "..", *args
      system "make", "install"
      # Remove stray ncdump executable as it conflicts with NetCDF.
      rm (bin+"ncdump")
    end

However, because it doesn't delete the "ncgen" executable, we still get the conflict the the hdf4 stays unlinked. 

I deleted it by hand after the installation, and then did the link, which finished ok. So this is a sure fix that you guys can implement. Here: 

Cheers
Henrique

$ brew install hdf4
==> Installing hdf4 from homebrew/versions
==> Downloading http://www.hdfgroup.org/ftp/HDF/releases/HDF4.2.11/src/hdf-4.2.11.tar.bz2
Already downloaded: /Users/hbarbosa/Library/Caches/Homebrew/hdf4-4.2.11.tar.bz2
==> Patching
patching file CMakeLists.txt
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/hdf4/4.2.11 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VER
==> make install
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/ncgen
Target /usr/local/bin/ncgen
is a symlink belonging to netcdf. You can unlink it:
  brew unlink netcdf

To force the link and overwrite all conflicting files:
  brew link --overwrite hdf4

To list all files that would be deleted:
  brew link --overwrite --dry-run hdf4

Possible conflicting files are:
/usr/local/bin/ncgen -> /usr/local/Cellar/netcdf/4.3.3.1_4/bin/ncgen

$ rm -f /usr/local/Cellar/hdf4/4.2.11/bin/ncgen
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
    hdf4
brew link hdf4

$ brew link hdf4
Linking /usr/local/Cellar/hdf4/4.2.11... 2080 symlinks created
Henriques-MacBook-Air:uspdata hbarbosa$ brew doctor
Your system is ready to brew.

### PR DESCRIPTION
**Pull request by hmjbarbosa**
Problem: installing hdf4 conflicts with installed netcdf.

Reason: hdf4 tries to link /usr/local/bin/ncgen which is already provided by netcdf.

How to reproduce: brew install netcdf, then brew install hdf4.

Solution: remove hcgen from Cellar/hdf4 folder so that it is not linked. This is the same at currently done for bin/ncdump. 

Fixes #1325.

### CODE REVIEW

File: hdf4.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ def install
 
       # Remove stray ncdump executable as it conflicts with NetCDF.
```
Comment (MikeMcQuaid) [inline hdf4.rb]: Can you update this comment? Thanks!

**Comment by MikeMcQuaid** [comment]
Great work! One nit and we'll ship this.

File: hdf4.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline hdf4.rb]: @hmjbarbosa ping on this.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
Seems like this should include a formula revision bump, since it changes the set of files which get installed (and should be included in the bottle).

**Comment by MikeMcQuaid** [comment]
Merged in https://github.com/Homebrew/homebrew-versions/pull/1348.

**Pull request closed without merging**